var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -196,16 +196,12 @@ def check_install(self):
             with open("check.c", 'w') as f:
                 f.write(source)
             if '+mpi' in spec:
-                cc = which('%s' % spec['mpi'].mpicc)
+                cc = which(spec['mpi'].mpicc)
             else:
-                cc = which('cc')
-            # TODO: Automate these path and library settings
-            cc('-c', "-I%s" % join_path(spec.prefix, "include"), "check.c")
-            cc('-o', "check", "check.o",
-               "-L%s" % join_path(spec.prefix, "lib"),
-               "-L%s" % join_path(spec.prefix, "lib64"),
-               "-lhdf5",
-               "-lz")
+                cc = which(self.compiler.cc)
+            cc(*(['-c', "check.c"] + spec['hdf5'].cppflags.split()))
+            cc(*(['-o', "check", "check.o"] +
+                 spec['hdf5'].libs.ld_flags.split() ))
             try:
```

Comment:
Without the `try`, how to I prevent the error from aborting Spack? I want to output both the current and the desired test output so that one sees what went wrong. Since I'm capturing the output anyway, wouldn't that show any error message?